### PR TITLE
Big Refactor

### LINKED ISSUE #41: Commands Clean Up & Local Dev Testing
commands.rs is getting unwieldly with the changes to the commands, might be better to seperate out some of the commands into their own specific files as an example:
```
commands.rs
commands/
- queue.rs
```
Also, to improve local testing, a new env variable for the prefix

### LINKED ISSUE #47: Sunny tries to leave calls twice
`!leave` gets sunny to try to leave twice, seems to be the result of the `voice_state_update(...)` handler.

### PR DESCRIPTION
**Pull request by NULLx76**
Refactored commands to extract the core logic to be handled in the effects, this allows us to be able to re-use this logic in other places where it was already being manually re-used anyway.

Added custom Sunny error/result handling

Added prefix to the .env for local testing

wip #41

Co-authored-by: Victor Roest <victor@xirion.net>

### CODE REVIEW

File: src/effects/queue.rs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    let mut page: usize = 0;
+
+    // await interactions i.e. button presses
+    let mut collector = message
+        .await_component_interactions(&ctx.shard)
+        .timeout(Duration::from_secs(3600)) // 1h
+        .await;
+
+    // Process button presses
+    while let Some(mci) = collector.next().await {
```
Comment (NULLx76) [inline src/effects/queue.rs:179]: Should we live update the queue every `t`?

**Comment by Druue** [comment]
I noticed there was some panic thrown when stopping sunny earlier with ctrl-c should test that before merging

**Comment by NULLx76** [comment]
> I noticed there was some panic thrown when stopping sunny earlier with ctrl-c should test that before merging

Maybe we should just try and eliminate all `.unwrap()`s then

**Comment by jonay2000** [comment]
> > I noticed there was some panic thrown when stopping sunny earlier with ctrl-c should test that before merging
> 
> Maybe we should just try and eliminate all `.unwrap()`s then

working on it

**Comment by jdonszelmann** [comment]
> > I noticed there was some panic thrown when stopping sunny earlier with ctrl-c should test that before merging
> 
> Maybe we should just try and eliminate all `.unwrap()`s then

working on it

File: src/effects/mod.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+pub use leave::leave;
+pub use play::play;
+pub use skip::skip;
+pub use stop::stop;
+
+use songbird::input::Metadata;
+use std::time::Duration;
+
+/// `split_duration` splits a [`Duration`] into a (minutes, seconds) tuple
+pub const fn split_duration(d: Duration) -> (u64, u64) {
```
Comment (Druue) [inline src/effects/mod.rs]: Do we want these to be public? They're only used within effects/

Comment (jdonszelmann) [reply src/effects/mod.rs]: `pub (crate)` ?

Comment (Druue) [reply src/effects/mod.rs]: Should be fine having them private right? They're still accessible to all child modules that way

File: src/effects/queue.rs (hunk 1 above)
Comment (Druue) [reply src/effects/queue.rs:179]: How do you mean?

File: src/effects/mod.rs (hunk 2 above)
Comment (jdonszelmann) [reply src/effects/mod.rs]: if you only use it in children you indeed can make it private

**Comment by Druue** [comment]
New issue seems to have sprung up where calling `leave` will both successfully have sunny leave and respond "Left voice" but will also log an error
image

File: src/effects/queue.rs (hunk 1 above)
Comment (Druue) [reply src/effects/queue.rs:179]: Gonna leave this for now, per dm

**Pull request merged**